### PR TITLE
Added STJ tests for deserializing null values

### PR DESCRIPTION
**Pull request by prathmesh-stripe**
### Why?
Add some tests as followup to https://github.com/stripe/stripe-dotnet/pull/3223

### What?
- Add test to serialize and deserialize basic object with System.Text.Json
- Add serialization+deserialization test for Newtonsoft that has null fields assigned to non-nullable fields.

### CODE REVIEW

File: src/StripeTests/Infrastructure/SerializationTest.cs [hunk 1]
```diff
@@ -44,6 +44,14 @@ public void Serialize()
                 JsonConvert.DeserializeObject<Event>(serialized),
                 Formatting.Indented);
             Assert.Equal(serialized, reserialized);
+
+            json = GetResourceAsString("api_fixtures.customer_with_expansions.json");
```
Comment (prathmesh-stripe) [inline src/StripeTests/Infrastructure/SerializationTest.cs:48]: `description` is nullable in the JSON, required primitive in Customer class. This is handled gracefully by Newtonsoft. This same test does not pass with SystemTextJson.

Comment (jar-stripe) [reply src/StripeTests/Infrastructure/SerializationTest.cs:48]: this confuses me because strings should work like objects, not other primatives.  do we know why this is failing?

File: src/StripeTests/Infrastructure/StjDeserializationTest.cs [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+            {
+                Date = roundedDate,
+            };
+
+            var reloaded = JsonSerializer.Deserialize<TestObjectDateTime>(JsonSerializer.Serialize(obj));
+            Assert.Equal(reloaded.Date, obj.Date);
+        }
+
+        [Fact]
+        public void HandleNull()
```
Comment (jar-stripe) [inline src/StripeTests/Infrastructure/StjDeserializationTest.cs:28]: i think we should move this into https://github.com/stripe/stripe-dotnet/pull/3158/files

File: src/StripeTests/Infrastructure/SerializationTest.cs (hunk 1 above)
Comment (prathmesh-stripe) [reply src/StripeTests/Infrastructure/SerializationTest.cs:48]: This is passing with Newtonsoft. Skipped it in STJ test

File: src/StripeTests/Infrastructure/StjDeserializationTest.cs (hunk 2 above)
Comment (prathmesh-stripe) [reply src/StripeTests/Infrastructure/StjDeserializationTest.cs:28]: But this is a passing test so just wanted to get this merged.
